### PR TITLE
Add ContextualReflection tests

### PR DESCRIPTION
**Pull request by sdmaclea**
Adds tests for #36845 and the context sensitive reflection APIs

### CODE REVIEW

File: src/System.Runtime.Loader/tests/ContextualReflection.cs [hunk 1]
```diff
@@ -0,0 +1,1055 @@
...
+        public AssemblyLoadContext isolatedAlc { get; }
+        public Assembly isolatedAlcAssembly { get; }
+        public Type isolatedAlcFixtureType { get; }
+        public IContextualReflectionTestFixture isolatedAlcFixtureInstance { get; }
+
+        public AssemblyLoadContext defaultAlc { get; }
+        public Assembly defaultAlcAssembly { get; }
+        public Type defaultAlcFixtureType { get; }
+
+        [MethodImplAttribute(MethodImplOptions.NoInlining)]
```
Comment (jkotas) [inline src/System.Runtime.Loader/tests/ContextualReflection.cs]: It should not be necessary to mark these with NoInlining attribute. `CORINFO_FLG_DONT_INLINE_CALLER` flag in the JIT should ensure that these are not getting inlined.

File: src/System.Runtime.Loader/tests/ContextualReflectionDependency/Configurations.props [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿<Project DefaultTargets="Build">
+  <PropertyGroup>
+    <BuildConfigurations>
+      netstandard;
```
Comment (jkotas) [inline src/System.Runtime.Loader/tests/ContextualReflectionDependency/Configurations.props]: This depends on netcoreapp-specific APIs. It should not be building for netstandard.

File: src/System.Runtime.Loader/tests/ContextualReflectionDependency/Configurations.props [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿<Project DefaultTargets="Build">
+  <PropertyGroup>
+    <BuildConfigurations>
+      netcoreapp-Unix;
+      netcoreapp-Windows_NT;
```
Comment (jkotas) [inline src/System.Runtime.Loader/tests/ContextualReflectionDependency/Configurations.props]: This does not have any OS-specific code. It does not need to be build OS specific. (Also, we prefer to not have OS-specific builds for tests.)

Comment (sdmaclea) [reply src/System.Runtime.Loader/tests/ContextualReflectionDependency/Configurations.props]: I was copying from 
https://github.com/dotnet/corefx/blob/8b515adcd4ea0b8233f018f9e87ef946ce0e7c42/src/System.Runtime.Loader/tests/Configurations.props#L4-L5

I'll fix this library.

**Comment by sdmaclea** [comment]
Unless there are objections, I'll merge this.

**Pull request merged**